core/unix/os.c

```diff
@@ -9360,13 +9360,21 @@ get_memory_info_from_os(const byte *pc, byte **base_pc, size_t *size,
 extern void deadlock_avoidance_unlock(mutex_t *lock, bool ownable);
 
 void
-mutex_wait_contended_lock(mutex_t *lock)
+mutex_wait_contended_lock(mutex_t *lock, priv_mcontext_t *mc)
 {
 #ifdef CLIENT_INTERFACE
     dcontext_t *dcontext = get_thread_private_dcontext();
     bool set_client_safe_for_synch =
                       ((dcontext != NULL) && IS_CLIENT_THREAD(dcontext) &&
                         ((mutex_t *)dcontext->client_data->client_grab_mutex == lock));
+    if (mc != NULL) {
+        ASSERT(dcontext != NULL);
+        *get_mcontext(dcontext) = *mc;
```

Comment:
Hmm if the caller did not set DR_MC_MULTIMEDIA this will clobber xmm fields with uninitialized data.  Either DR_MC_ALL needs to be required in the API routine, or dr_mcontext_to_priv_mcontext() needs to be used somewhere.  Unfortunately we never merged the formats which makes for no zero-cost solution here other than passing in dr_mcontext_t instead (prob under _IF_CLIENT_INTERFACE()).

Reply:
I vote for requiring DR_MC_ALL and asserting for it in dr_app_recurlock_lock